executor-scripts/linux/vrrp

```diff
@@ -0,0 +1,358 @@
+#!/bin/sh
+# This executor is responsible for setting up the Virtual Router Redundancy Protocol (VRRP) overlay interfaces.
+#
+# Copyright (C) 2022 EasyNetDev <devel@easynet.dev>
+#
+# Permission to use, copy, modify, and/or distribute this software for any
+# purpose with or without fee is hereby granted, provided that the above
+# copyright notice and this permission notice appear in all copies.
+#
+# This software is provided 'as is' and without any warranty, express or
+# implied. In no event shall the authors be liable for any damages arising
+# from the use of this software.
+#
+# Sun, 21 Apr 2023 10:36:31 +0300
+#  -- EasyNetDev <devel@easynet.dev>
+#
+# Move the crc32sum tool to internal applet of ifupdown-ng and change the VRRP 
+# executor to use the new applet instead external crc32sum tool.
+# The new applet is called ifcrc32sum.
+#
+# Sun, 03 Jan 2024 19:00:26 +0300
+#  -- EasyNetDev <devel@easynet.dev>
+#
+# This executor will do the configuration for MACVLAN virtual interfaces.
+#
+# Each MACVLAN interface must be controlled by a VRRP daemon, like FRRouting.
```

Comment:
If a MACVLAN interface has to be controlled by a VRRP daemon, then why is it ifupdown-ng’s responsibility to set up the interface?

The interface should be managed by the daemon which is supervising it, such as FRR.

Reply:
Then you don't understand how VRRP is working in Linux.

Let me explain: FRRouting vrrpd daemon is just controlling protodown on/off of the MACVLAN interfaces. FRRouting DOESN'T create these interfaces. Is just matching the MAC address of MACVLAN interface and is adding in its stack to control the interface.

That's why we need to create these interfaces in OS level.

Hope now is clear.

Reply:
Yes, I actually do understand how VRRP is working in Linux.

What I am implying is that this contract between interface and manager is presently wrong.  I am not convinced that we should be creating interfaces for other managers to manage.

Those managers should create their own interfaces if they are otherwise managing the lifecycle of the interface.

Reply:
Unfortunately is not their job. They need these interfaces to be configured by OS. FRRouting has the documentation for VRRP where explicit says that interfaces must be configured by OS.
ifupdown2 has such executors that are doing this. In documentation they mention about this feature from ifupdown2.